ultralytics/nn/autobackend.py

```diff
@@ -393,7 +398,30 @@ def forward(self, im, augment=False, visualize=False, embed=None):
             y = self.session.run(self.output_names, {self.session.get_inputs()[0].name: im})
         elif self.xml:  # OpenVINO
             im = im.cpu().numpy()  # FP32
-            y = list(self.ov_compiled_model(im).values())
+
+            inference_mode = "throughput"  # either 'latency' or 'throughput'
+            # Latency optimized inference at batch-size 1
+            if inference_mode == "latency":
+                y = list(self.ov_compiled_model(im).values())
+
+            # Throughput optimized inference using OpenVINO AsyncInferQueue
+            elif inference_mode == "throughput":
+                from openvino.runtime import AsyncInferQueue
+
+                results = []  # this list will be filled by the callback function
+
+                def callback(request, userdata):
+                    """Callback function to handle the completion of an async inference request."""
+                    results.append(request.results)  # directly append the inference result to 'results'
```

Comment:
The order here is not guaranteed, so let's use userdata parameter to pass the number of the request.

Reply:
``` 
                for i, image in enumerate(im):
                    async_queue.start_async(inputs={self.input_name: image[None]}, userdata=i) 
```

Reply:
Great point! Fixed in https://github.com/ultralytics/ultralytics/pull/8058/commits/f9f8b1ee9e98006bb43446f9906538eeaadfdd86

Reply:
But this is only part of the solution. You also need to handle userdata in the callback function to ensure you merge results in the proper order (you cannot just append to list one by one)

Reply:
@adrianboguszewski ah got it! Ok I'll look into this.

Reply:
All done in https://github.com/ultralytics/ultralytics/pull/8058/commits/f19995edf49c94cd4697f9d3bf419f9fe95396b8